core/results/get-overall-outcome.ts

```diff
@@ -0,0 +1,24 @@
+import { IResultWithOutcome } from "./result-with-outcome.i";
+import { TestOutcome } from "./test-outcome";
+
+export function getOverallOutcome(results: Array<IResultWithOutcome>) {
+  const outcomes = results.map(result => result.outcome);
+
+  if (outcomesContains(outcomes, TestOutcome.Error)) {
+    return TestOutcome.Error;
+  }
+
+  if (outcomesContains(outcomes, TestOutcome.Fail)) {
+    return TestOutcome.Fail;
+  }
+
+  if (outcomesContains(outcomes, TestOutcome.Pass)) {
+    return TestOutcome.Pass;
+  }
+
+  return TestOutcome.Skip;
+}
+
+function outcomesContains(outcomes: Array<TestOutcome>, outcome: TestOutcome) {
+  return outcomes.some(o => o === outcome);
```

Comment:
It looks as though [`indexOf` is faster than `some`](https://jsperf.com/indexof-vs-somef) - why not use that?

Reply:
More readable I think unless you disagree?

Reply:
+ I've just benchmarked the function in question on node over a billion iterations ran 10 times shows only a 15% slowdown using some. Perhaps browsers are more affected?

Reply:
For context the average time for execution is ~ 160ns (yep nano :) ) I'm not sure we'd manage to gain enough performance for the readability

Reply:
I think that `arr.indexOf(x) !== -1` is quicker to glance at and see what's going on but that's probably only because it's syntax that I've used for years while `arr.some(i => i === x)` is fairly new.

Happy to keep `some` though

Reply:
@7jpsan thanks for you input :) good to see you on here!!